src/data/arena-tags/utils/load-arena-tag.ts

```diff
@@ -0,0 +1,13 @@
+import type { ArenaTag } from "#arena-tags/arena-tag";
+import { getArenaTag } from "#arena-tags/utils/get-arena-tag";
+
+/**
+ * When given a battler tag or json representing one, creates an actual ArenaTag object with the same data.
+ * @param source - The source {@linkcode ArenaTag}
+ * @returns The valid {@linkcode ArenaTag}
+ */
+export function loadArenaTag(source: ArenaTag | any): ArenaTag | undefined {
+  const tag = getArenaTag(source.tagType, source.sourceId, source.turnCount, source.sourceMoveId, source.side);
+  tag?.loadTag(source);
+  return tag;
+}
```

Comment:
I'm not a fan of allowing `any`


```suggestion
export function loadArenaTag(source: ArenaTag): ArenaTag | undefined {
  const tag = getArenaTag(source.tagType, source.sourceId, source.turnCount, source.sourceMoveId, source.side);
  tag?.loadTag(source);
  return tag;
}
```

Reply:
`any` is required (at least atm) because this may directly take a JSON as input. So this is probably out of this PR's scope.